src/backend/distributed/utils/resource_lock.c

```diff
@@ -248,7 +250,10 @@ lock_shard_resources(PG_FUNCTION_ARGS)
 			continue;
 		}
 
-		EnsureTablePermissions(relationId, aclMask);
+		if (!SkipAdvisoryLockPermissionChecks)
+		{
+			EnsureTablePermissions(relationId, aclMask);
```

Comment:
In an ideal world, `LockShardResource` should be doing `EnsureTablePermissions()`. But, for now at least do it for `lock_shard_resources `, which we can expand to `LockShardResource`

Reply:
What would be a good time to do that?